syntax = "proto3";

service Communicator {
    rpc Customer_Process (Info_Request) returns (Info_Response) {}
    rpc BranchInfo (BroadCast) returns (BroadCast_Response) {}
}

message Info_Request {
    float customer_id = 1;
    float money = 2;
    string interface = 3;
}

message Info_Response {
    float new_balance = 1;
    string process_result = 2;
}

message BroadCast{
    float branch_balance = 1;
    float branch_id = 2;
    string interface = 3;
}

message BroadCast_Response{
    float value = 1;
}